grammar.js

```diff
@@ -450,11 +450,17 @@ module.exports = grammar(C, {
     constructor_or_destructor_definition: $ => seq(
       repeat($._constructor_specifiers),
       field('declarator', $.function_declarator),
-      optional($.field_initializer_list),
       choice(
-        field('body', choice(
-          $.compound_statement,
-          $.try_statement)),
+        seq(
+          optional($.field_initializer_list),
+          field('body', $.compound_statement)
+        ),
+        seq(
+          optional('try'),
```

Comment:
What might actually make more sense is to make another `constructor_try_statement` that is aliased as a `try_statement`.  Something like:

```cpp
constructor_try_statement: $ => seq(
  'try',
  optional($.field_initializer_list),
  field('body', $.compound_statement),
  repeat1($.catch_clause)
),

constructor_or_destructor_definition: $ => seq(
  repeat($._constructor_specifiers),
  field('declarator', $.function_declarator),
  field('body', choice(
    seq(
      optional($.field_initializer_list),
      field('body', $.compound_statement)
    ),
    alias($.constructor_try_statement, $.try_statement)
  ))
)
```

This would let the parsed tree still have a "try_statement" (labeled as the "body") as a named node like it is now. What do you think?

Reply:
Thanks for checking the PR.

Yes, I agree that aliasing constructor_try_statement as try_statement is a good idea. I just tested that locally and it is highlighted in neovim as expected.

Please note one thing in your proposed change: for the usual `compound_statement` variant we now have two nested body fields and this results in `field_initializer_list` being labeled also as `body`:

```cpp
struct Test
{
    Test() : member(1) {}
    Test() try : member(5) {} catch (...) {}

struct_specifier [0, 0] - [10, 1]
  name: type_identifier [0, 7] - [0, 11]
  body: field_declaration_list [1, 0] - [10, 1]
    function_definition [2, 4] - [2, 25]
      declarator: function_declarator [2, 4] - [2, 10]
        declarator: identifier [2, 4] - [2, 8]
        parameters: parameter_list [2, 8] - [2, 10]
      body: field_initializer_list [2, 11] - [2, 22]
        field_initializer [2, 13] - [2, 22]
          field_identifier [2, 13] - [2, 19]
          argument_list [2, 19] - [2, 22]
            number_literal [2, 20] - [2, 21]
      body: compound_statement [2, 23] - [2, 25]
    function_definition [3, 4] - [3, 44]
      declarator: function_declarator [3, 4] - [3, 10]
        declarator: identifier [3, 4] - [3, 8]
        parameters: parameter_list [3, 8] - [3, 10]
      body: try_statement [3, 11] - [3, 44]
        field_initializer_list [3, 15] - [3, 26]
          field_initializer [3, 17] - [3, 26]
            field_identifier [3, 17] - [3, 23]
            argument_list [3, 23] - [3, 26]
              number_literal [3, 24] - [3, 25]
        body: compound_statement [3, 27] - [3, 29]
        catch_clause [3, 30] - [3, 44]
          parameters: parameter_list [3, 36] - [3, 41]
          body: compound_statement [3, 42] - [3, 44]
```

Before we introduced this whole function try block, c-tor `body` did not include initalizer list:
```js
    constructor_or_destructor_definition: $ => seq(
      repeat($._constructor_specifiers),
      field('declarator', $.function_declarator),
      optional($.field_initializer_list),
      choice(
        field('body', $.compound_statement),
        $.default_method_clause,
        $.delete_method_clause
      )
    ),
```
Is this change intentional?

Reply:
Nope, it wasn't :)

You can just take out labelling the outer choice as 'body' and we'd get the desired outcome, I believe.

Reply:
Fixed as you suggested, thanks!